### PR TITLE
Update changelog for the upcoming v0.11 release

### PR DESCRIPTION
**Pull request by paullo0106**
## Proposed changes

Update changelog for the upcoming 0.11 release

## Types of changes

What types of changes does your code introduce to **CausalML**?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation Update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/uber/causalml/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/uber/causalml)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

NA

### CODE REVIEW

File: docs/changelog.rst [hunk 1]
```diff
@@ -2,6 +2,31 @@
...
+0.11.0 (2021-07-28)
+------------------
+- CausalML surpassed `2K stars <https://github.com/uber/causalml/stargazers>`_!
+- We have 3 new community contributors, Jannik (`@jroessler <https://github.com/jroessler>`_), Mohamed (`@ibraaaa <https://github.com/ibraaaa>`_), and Leo (`@lleiou <https://github.com/lleiou>`_). Thanks for the contribution!
+
+Major Updates
+~~~~~~~~~~~~~
+- Make tensorflow dependency optional by @jeongyoonlee (`#343 <https://github.com/uber/causalml/pull/343>`_)
+- Add delta-delta-p (ddp) tree inference approach by @jroessler (`#327 <https://github.com/uber/causalml/pull/327>`_)
+- Add conda env files for Python 3.6, 3.7, and 3.8 by @jeongyoonlee (`#324 <https://github.com/uber/causalml/pull/324>`_)
```
Comment (jeongyoonlee) [inline docs/changelog.rst:14]: and Python 3.9 :)

Comment (paullo0106) [reply docs/changelog.rst:14]: Thanks! I added that in the first #343 description - does this look good?

**Review by jeongyoonlee — APPROVED** [review]
LGTM. Thanks!

**Review by ppstacy — APPROVED** [review]
lgtm

**Pull request merged**